src/sonic_ax_impl/mibs/ieee802_1ab.py

```diff
@@ -511,28 +510,39 @@ def update_rem_if_mgmt(self, if_oid, if_name):
             if len(mgmt_ip_str) == 0:
                 # the peer advertise an emtpy mgmt address
                 return
-            time_mark = int(lldp_kvs['lldp_rem_time_mark'])
-            remote_index = int(lldp_kvs['lldp_rem_index'])
-            subtype = self.get_subtype(mgmt_ip_str)
-            ip_hex = self.get_ip_hex(mgmt_ip_str, subtype)
-            if subtype == ManAddrConst.man_addr_subtype_ipv4:
-                addr_subtype_sub_oid = 4
-                mgmt_ip_sub_oid = (addr_subtype_sub_oid, *[int(i) for i in mgmt_ip_str.split('.')])
-            elif subtype == ManAddrConst.man_addr_subtype_ipv6:
-                addr_subtype_sub_oid = 6
-                mgmt_ip_sub_oid = (addr_subtype_sub_oid, *[int(i, 16) if i else 0 for i in mgmt_ip_str.split(':')])
-            else:
-                logger.warning("Invalid management IP {}".format(mgmt_ip_str))
-                return
-            self.if_range.append((time_mark,
-                                  if_oid,
-                                  remote_index,
-                                  subtype,
-                                  *mgmt_ip_sub_oid))
-
-            self.mgmt_ips.update({if_name: {"ip_str": mgmt_ip_str,
-                                            "addr_subtype": subtype,
-                                            "addr_hex": ip_hex}})
+            
+            mgmt_ip_set=set()
+            for mgmt_ip in mgmt_ip_str.split(','):
+                time_mark = int(lldp_kvs['lldp_rem_time_mark'])
+                remote_index = int(lldp_kvs['lldp_rem_index'])
+                subtype, exploded_mgmt_ip = self.get_subtype_and_exploded_ip(mgmt_ip)
```

Comment:
>get_subtype_and_exploded_ip [](start = 49, length = 27)

The code is okay, I am think about more readable and concise.
I found some sample code: [https://github.com/Azure/sonic-snmpagent/blob/9b83459d647622a12387085f01d2256fd69430e2/src/sonic_ax_impl/mibs/vendor/cisco/bgp4.py#L41-L46](https://github.com/Azure/sonic-snmpagent/blob/9b83459d647622a12387085f01d2256fd69430e2/src/sonic_ax_impl/mibs/vendor/cisco/bgp4.py#L41-L46)

Suggest you extract a common method and reuse.

Reply:
Thanks @qiluo-msft . Updated to use updated utility API.